### ISSUE TITLE
About Trainning Dataset

### ISSUE THREAD

**Issue by YongLD** [ryohachiuma/Deep-6dPose#1]
Thanks for your amazing work for 6D pose estimation! I have perusal your paper but found that I can't train the model in LINEMOD dataset. Can you tell me how to preprocess the LINEMOD dataset with your code?

**Comment by ryohachiuma** [comment]
I'm sorry for disappointing you, but this is not the official implementation. 
I implemented just for studying. 
If you see the code generateData.py, you can re-write it to adapt with another dataset.

**Comment by YongLD** [comment]
> I'm sorry for disappointing you, but this is not the official implementation.
> I implemented just for studying.
> If you see the code generateData.py, you can re-write it to adapt with another dataset.

No problem, thanks for your response, did you use tensorflow_1.3.0?

**Comment by ryohachiuma** [comment]
I followed the setup of Mask R-CNN, so the version of tensorflow is 1.3.0 as you said!

**Comment by YongLD** [comment]
> I followed the setup of Mask R-CNN, so the version of tensorflow is 1.3.0 as you said!

HaHa, thank you very much. I have trained and tested the model in ycb and linemod dataset but found a terrible result. Did you find a nice result when testing the model in ycb dataset?

**Comment by ryohachiuma** [comment]
Thank you for using it!
It was a long time ago but I'm sure that I got fine results with ycb dataset (not with linemod...)

How were your loss graphs?
The segmentation mask and detection bounding box are also worse?

**Comment by YongLD** [comment]
> Thank you for using it!
> It was a long time ago but I'm sure that I got fine results with ycb dataset (not with linemod...)
> 
> How were your loss graphs?
> The segmentation mask and detection bounding box are also worse?
Yeah~~ I didn't calculate the loss of mask and detection bounding box but only evaluate the pose with evaluation criteria 5cm5`, ADD and ADI.
I don't know what's the problem.
I really hope to evaluate your pretrained model in this evaluation criteria if your model is still holding.

**Comment by YongLD** [comment]
Yeah~~ I didn't calculate the loss of mask and detection bounding box but only evaluate the pose with evaluation criteria 5cm5`, ADD and ADI.
I don't know what's the problem.
I really hope to evaluate your pretrained model in this evaluation criteria if your model is still holding.

**Comment by YongLD** [comment]
@kemangjaka 
Yeah~~ I didn't calculate the loss of mask and detection bounding box but only evaluate the pose with evaluation criteria 5cm5`, ADD and ADI.
I don't know what's the problem.
I really hope to evaluate your pretrained model in this evaluation criteria if your model is still holding.

**Comment by ryohachiuma** [comment]
What do you mean you didn't calculate the loss of mask and detection?
You didn't train them or you didn't visualize them?
The tensorboard will visualize the loss graphs.

**Comment by YongLD** [comment]
@kemangjaka 
oh haha I so sorry! I find a nice result of segmentation and bounding box in loss graphs.
And I find a little change in my code that I didn't train the stage 2 and stage 3 and stage 4 when I find that promote tinily. Did it have a lot of influence?

**Comment by ryohachiuma** [comment]
Ok. How was the loss graph of pose? The pose loss also decrease like the others?
As the domain of mscoco and ycb dataset is different each other, the stage 2, 3 and 4 are important.

**Comment by YongLD** [comment]
@kemangjaka 
Amazing!! Maybe that's the problem.
pose loss also decreases like the others with convergence to loss 0.6(terrible), even though the loss graphs oscillate up and down.

**Comment by ryohachiuma** [comment]
Hmm...
You already visualized the ground truth quaternion and translation (z value) is correct, right?
I think the pose estimation is a difficult task much more than segmentation and detection. 
Therefore, after the segmentation and detection loss converged, the pose loss would begin to decrease. 

I will also train the model in my environment!

**Comment by YongLD** [comment]
HAHA~ Appreciate you! 6D pose is a challenge with RGB-only.
I will train with stage 2,3,4 for more improvement.

**Comment by ryohachiuma** [comment]
Let me know if you couldn't decrease the loss with full training!

**Comment by YongLD** [comment]
ok~ I will!

**Comment by YongLD** [comment]
@kemangjaka 
OK~ I will!

**Comment by YongLD** [comment]
@kemangjaka 
 I come again haha! How many iterations did you set for 1/2/3/4 stage?  Is it the same as that in the code?

**Comment by ryohachiuma** [comment]
Hi!
I uploaded the last version so that it is same in the code~
It doesn't work?

**Comment by YongLD** [comment]
@kemangjaka 
HiHiHi~
It works but I think the iterations should be larger for the better convergence. Is there any relationship between the four stages when I must increase the iterations?

**Comment by ryohachiuma** [comment]
@YongLD 
Hi, sorry for the slow reply (I was busy with writing the paper...)
How do you do?
Did it worked...?

**Comment by YongLD** [comment]
@kemangjaka 
That's ok~ I know that haha~
I still find the bad result of ADD and ADI in ycb dataset, possibly because of the occlusion. 
I didn't change a lot on your code but the configuration. 
Could you provide your trained model for me? Then I can compare the result between yours and mine~

**Comment by YongLD** [comment]
@kemangjaka 
The largest trouble is that when the mrcnn_pose_loss was down to 0.1, the val_pose_loss still at 5.0 without a decrease(overfitting?). I don't know why. Is it probably because of batch normalization?

**Comment by YongLD** [comment]
I have good new HAHA!!!
Now the val_pose_loss decrease at 0.15, I find it will be a great result in ADD and ADI~
Thank you very much! I sorry to trouble you for so long~~

**Comment by YongLD** [comment]
@kemangjaka 
I have good new HAHA!!!
Now the val_pose_loss decrease at 0.15, I find it will be a great result in ADD and ADI~
Thank you very much! I sorry to trouble you for so long~~
